crates/hir-ty/src/infer/unify.rs

```diff
@@ -252,6 +256,51 @@ impl<'a> InferenceTable<'a> {
         }
     }
 
+    /// Create a short-lived `InferenceTable` that used for doing chalk things for `t`,
+    /// without `InferenceContext`.
+    pub(crate) fn new_for_instant_use<T: TypeVisitable<Interner>>(
```

Comment:
Few nits around documentation/naming:

```suggestion
    /// Create a short-lived `InferenceTable` without an `InferenceTable` for doing Chalk things with `&T`.
    ///
    /// For context as to why this is necessary, see https://github.com/rust-lang/rust-analyzer/pull/17882.
    pub(crate) fn new_for_immediate_use<T: TypeVisitable<Interner>>(
```

I'm afraid of hir-ty, so any additional documentation would be nice!

I also proposed a rename; I _think_ that "immediate" is a bit closer to what you're looking for :)

Reply:
"immediate" sounds much better. Thanks!